data/events/scripts/monster.lua

```diff
@@ -0,0 +1,52 @@
+function Monster:onDropLoot(corpse)
+	if (configManager.getNumber(configKeys.RATE_LOOT) == 0) then
+		corpse:decay()
+		return
+	end
+	local player = Player(corpse:getCorpseOwner())
+	local mType = self:getType()
+	if not player or player:getStamina() > 840 then
+		local monsterLoot = mType:getLoot()
+		for it = 1, #monsterLoot do
+			local loot = monsterLoot[it]
+			local itemList = createLootItem(monsterLoot[it])
+			if (#itemList > 0) then
+				for i = 1, #itemList do
+					local item = itemList[i]
+					-- check containers
+					if item:isContainer() then
+						if not item:createLootContainer(loot.childLoot) then
```

Comment:
In the original C++ code, the container item would be deleted from memory. You're just forgetting about it and leaving the object around creating a memory leak.

Reply:
oopsie, thanks joe for your revision.